File: backend/dacha/src/main/java/io/featurehub/dacha/Application.java
package io.featurehub.dacha;

import cd.connect.lifecycle.ApplicationLifecycleManager;
import cd.connect.lifecycle.LifecycleStatus;
import io.featurehub.health.MetricsHealthRegistration;
import io.featurehub.jersey.FeatureHubJerseyHost;
import io.featurehub.lifecycle.TelemetryFeature;
import io.featurehub.publish.NATSFeature;
import org.glassfish.jersey.server.ResourceConfig;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import static io.featurehub.rest.Info.APPLICATION_NAME_PROPERTY;

public class Application {
  private static final Logger log = LoggerFactory.getLogger(Application.class);

  private static void initializeCommonJerseyLayer() {
    // register our resources, try and tag them as singleton as they are instantiated faster
    ResourceConfig config =
        new ResourceConfig(
                NATSFeature.class,
                TelemetryFeature.class,
                DachaFeature.class);

    // check if we should list on a different port
    MetricsHealthRegistration.Companion.registerMetrics(config);

    new FeatureHubJerseyHost(config).disallowWebHosting().start();

    log.info("Dacha Launched - (HTTP/2 payloads enabled!)");
  }

  public static void main(String[] args) {
    System.setProperty("user.timezone", "UTC");
    System.setProperty(APPLICATION_NAME_PROPERTY, "dacha");

    try {
      initializeCommonJerseyLayer();
      log.info("Cache has started");

      Thread.currentThread().join();
    } catch (Exception e) {
      log.error("Failed to start", e);
      ApplicationLifecycleManager.updateStatus(LifecycleStatus.TERMINATING);
      System.exit(-1);
    }
  }
}

File: backend/dacha/src/main/kotlin/io/featurehub/dacha/Application.kt
package io.featurehub.dacha

import cd.connect.lifecycle.ApplicationLifecycleManager
import cd.connect.lifecycle.LifecycleStatus
import io.featurehub.events.CloudEventsFeature
import io.featurehub.health.MetricsHealthRegistration
import io.featurehub.jersey.FeatureHubJerseyHost
import io.featurehub.lifecycle.TelemetryFeature
import io.featurehub.publish.NATSFeature
import io.featurehub.rest.Info
import org.glassfish.jersey.server.ResourceConfig
import org.slf4j.LoggerFactory

object Application {
  private val log = LoggerFactory.getLogger(Application::class.java)
  private fun initializeCommonJerseyLayer() {
    // register our resources, try and tag them as singleton as they are instantiated faster
    val config = ResourceConfig(
      NATSFeature::class.java,
      TelemetryFeature::class.java,
      DachaFeature::class.java,

      CloudEventsFeature::class.java
    )

    // check if we should list on a different port
    MetricsHealthRegistration.registerMetrics(config)
    FeatureHubJerseyHost(config).disallowWebHosting().start()
    log.info("Dacha Launched - (HTTP/2 payloads enabled!)")
  }

  @JvmStatic
  fun main(args: Array<String>) {
    System.setProperty("user.timezone", "UTC")
    System.setProperty(Info.APPLICATION_NAME_PROPERTY, "dacha")
    try {
      initializeCommonJerseyLayer()
      log.info("Cache has started")
      Thread.currentThread().join()
    } catch (e: Exception) {
      log.error("Failed to start", e)
      ApplicationLifecycleManager.updateStatus(LifecycleStatus.TERMINATING)
      System.exit(-1)
    }
  }
}
